csp_billing_adapter_microsoft/plugin.py

```diff
@@ -81,55 +79,75 @@ def get_account_info(config: Config):
     The information contains the metadata for compute and network.
     """
     account_info = _get_metadata()
-    account_info['compute'] = json.loads(account_info.get('compute', '{}'))
-    account_info['network'] = json.loads(account_info.get('network', '{}'))
     account_info['cloud_provider'] = get_csp_name(config)
 
     return account_info
 
 
-def get_api_header():
-    """
-    Get the header to be used in requests
+def _get_metadata():
+    """Return a dict containing compute, network and signature information."""
+    metadata = {}
+    try:
+        metadata = _get_instance_metadata()
+        metadata['attestedData'] = _get_signature()
+    except ValueError as error:
+        log.error('Could not load JSON from metadata: {}'.format(error))
+        for key in ['compute', 'network', 'signature', 'pkcs7']:
+            if not metadata.get(key):
+                metadata[key] = {}
+
+    return metadata
+
+
+def _get_instance_metadata():
+    """Return all compute and network information from metadata."""
+    api_latest_version = _get_latest_api_version()
+    return json.loads(
+        _fetch_metadata(
+            INSTANCE_INFO_URL.format(latest=api_latest_version)
+        )
```

Comment:
Constructs a multi-line code statement the reader is expected to follow there is really no advantage to creating the temporary assignment of `api_latest_version` plus the same issues exists as in the `_get_signature` function.